lib/chef-dk/command/generator_commands.rb

```diff
@@ -69,6 +69,12 @@ module SharedGeneratorOptions
         :description => "Use to set arbitrary attribute KEY to VALUE in the code_generator cookbook",
         :default => [],
         :proc => Proc.new { |s| generator_args <<  s.split("=").map(&:strip) }
+
+      option :branch,
+        :short => "-f BRANCH",
+        :long  => "--for BRANCH",
```

Comment:
I agree with others that `--pipeline` is probably better since it's more explicit. Also if this is not very common, we probably don't need a short option.